prov/efa/src/efa_mr.c

```diff
@@ -511,76 +521,67 @@ struct ibv_mr *efa_mr_reg_ibv_dmabuf_mr(struct ibv_pd *pd, uint64_t offset,
 static struct ibv_mr *efa_mr_reg_ibv_mr(struct efa_mr *efa_mr, struct fi_mr_attr *mr_attr,
 					int access, const uint64_t flags)
 {
-	if (flags & FI_MR_DMABUF)
-		return efa_mr_reg_ibv_dmabuf_mr(
-			efa_mr->domain->ibv_pd,
-			mr_attr->dmabuf->offset,
-			mr_attr->dmabuf->len,
-			(uintptr_t) mr_attr->dmabuf->base_addr + mr_attr->dmabuf->offset,
-			mr_attr->dmabuf->fd,
-			access
-		);
+	int ret, dmabuf_fd;
+	struct ibv_mr *mr = NULL;
+	uint64_t iova, offset;
+	size_t len;
+	enum efa_dmabuf_support_status *dmabuf_support;
+
+	dmabuf_support = &(efa_mr->domain->hmem_info[mr_attr->iface].dmabuf_support_status);
+
+	if (flags & FI_MR_DMABUF) {
+		goto dmabuf;
+	}
+
+	if (*dmabuf_support == EFA_DMABUF_NOT_SUPPORTED) {
+		goto no_dmabuf;
+	}
+dmabuf:
+	if (flags & FI_MR_DMABUF) {
+		dmabuf_fd = mr_attr->dmabuf->fd;
+		offset = mr_attr->dmabuf->offset;
+		iova = (uintptr_t) mr_attr->dmabuf->base_addr +
+		       mr_attr->dmabuf->offset;
+		len = mr_attr->dmabuf->len;
+	} else {
+		iova = (uint64_t) mr_attr->mr_iov->iov_base,
+		len = mr_attr->mr_iov->iov_len;
+		ret = ofi_hmem_get_dmabuf_fd(mr_attr->iface, (void *)iova, len,
+		                             &dmabuf_fd, &offset);
 
-	/*
-	 * TODO: remove the synapseai and neuron blocks by onboarding the
-	 * ofi_hmem_get_dmabuf_fd API.
-	 */
-#if HAVE_SYNAPSEAI
-	if (efa_mr_is_synapseai(efa_mr)) {
-		int dmabuf_fd;
-		uint64_t offset;
-		int ret;
-
-		ret = synapseai_get_dmabuf_fd(mr_attr->mr_iov->iov_base,
-						(uint64_t) mr_attr->mr_iov->iov_len,
-						&dmabuf_fd, &offset);
 		if (ret != FI_SUCCESS) {
-			EFA_WARN(FI_LOG_MR, "Unable to get dmabuf fd for Gaudi device buffer \n");
-			return NULL;
+			EFA_WARN(FI_LOG_MR,
+				 "Unable to get dmabuf fd for device buffer. "
```

Comment:
Add `iface`'s string  in the warning message

Reply:
Updated.